apps/server/src/modules/tldraw/service/tldraw.ws.service.spec.ts

```diff
@@ -297,170 +350,539 @@ describe('TldrawWSService', () => {
+				const pingSpy = jest.spyOn(ws, 'ping').mockImplementationOnce(() => {});
+				const sendSpy = jest.spyOn(service, 'send').mockImplementation(() => {});
+				const clearIntervalSpy = jest.spyOn(global, 'clearInterval');
+				jest.spyOn(Ioredis.Redis.prototype, 'subscribe').mockResolvedValueOnce({});
+
 				return {
 					messageHandlerSpy,
 					closeConnSpy,
+					pingSpy,
+					sendSpy,
+					clearIntervalSpy,
 				};
 			};
 
 			it('should close connection', async () => {
-				const { messageHandlerSpy, closeConnSpy } = await setup();
+				const { messageHandlerSpy, closeConnSpy, pingSpy, sendSpy, clearIntervalSpy } = await setup();
 
-				service.setupWSConnection(ws);
+				await service.setupWSConnection(ws, 'TEST');
 
-				await delay(10);
+				await delay(20);
 
 				expect(closeConnSpy).toHaveBeenCalled();
-
+				expect(clearIntervalSpy).toHaveBeenCalled();
 				ws.close();
 				messageHandlerSpy.mockRestore();
+				pingSpy.mockRestore();
 				closeConnSpy.mockRestore();
+				sendSpy.mockRestore();
+				clearIntervalSpy.mockRestore();
+			});
+		});
+
+		describe('when flushDocument failed', () => {
+			const setup = async () => {
+				ws = await TestConnection.setupWs(wsUrl, 'TEST');
+				const doc = TldrawWsFactory.createWsSharedDocDo();
+				doc.connections.set(ws, new Set<number>());
+
+				const flushDocumentSpy = jest.spyOn(boardRepo, 'flushDocument');
+				const errorLogSpy = jest.spyOn(logger, 'warning');
+
+				return {
+					doc,
+					flushDocumentSpy,
+					errorLogSpy,
+				};
+			};
+
+			it('should log error', async () => {
+				const { doc, flushDocumentSpy, errorLogSpy } = await setup();
+				flushDocumentSpy.mockRejectedValueOnce(new Error('error'));
+
+				await expect(service.closeConn(doc, ws)).rejects.toThrow('error');
+
+				expect(flushDocumentSpy).toHaveBeenCalled();
+				expect(errorLogSpy).toHaveBeenCalled();
+				ws.close();
 			});
 		});
 	});
 
+	describe('updateHandler', () => {
+		const setup = async (shouldPublishSpyThrowError: boolean) => {
+			ws = await TestConnection.setupWs(wsUrl);
+
+			const sendSpy = jest.spyOn(service, 'send').mockReturnValueOnce();
+			const errorLogSpy = jest.spyOn(logger, 'warning');
+			const publishSpy = jest.spyOn(Ioredis.Redis.prototype, 'publish');
+
+			const doc = TldrawWsFactory.createWsSharedDocDo();
+			const socketMock = TldrawWsFactory.createWebsocket(WebSocketReadyStateEnum.OPEN);
+			doc.connections.set(socketMock, new Set());
+			const msg = new Uint8Array([0]);
+
+			if (shouldPublishSpyThrowError) {
+				publishSpy.mockImplementationOnce((_channel, _message, cb) => {
+					if (cb) {
+						cb(new Error('error'));
+					}
+					return Promise.resolve(0);
+				});
+			} else {
+				publishSpy.mockResolvedValueOnce(1);
+			}
+
+			return {
+				doc,
+				sendSpy,
+				socketMock,
+				msg,
+				errorLogSpy,
+				publishSpy,
+			};
+		};
+
+		it('should call send method', async () => {
+			const { sendSpy, doc, socketMock, msg } = await setup(false);
```

Comment:
Remove false and create 2 different setups, with only parts that are really needed.

Some setups / its in this file are not correct.